tests/unit/test_charm.py

```diff
@@ -85,6 +83,7 @@ def setUp(self):
 
     @responses.activate
     def test_remote_write_configuration(self):
+        self.harness.set_can_connect("agent", True)
```

Comment:
The `setUp` calls `begin_with_initial_hooks`, so it seems natural to me to add `self.harness.container_pebble_ready("agent")` immediately after, instead of `set_can_connect(True)`.
A test that specifically tests for False could do `set_can_connect(False)` in that one place even when we have `container_pebble_ready("agent")` at the top.

Reply:
good catch!